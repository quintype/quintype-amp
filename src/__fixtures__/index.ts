import { config, publisherConfig, ampConfig, configOpts } from "./config.fixture";
import { textStory } from "./text-story.fixture";
import { storyWithManyJsEmbeds } from "./story-with-jsembeds.fixture";
import { relatedStories } from "./related-stories";
import { seo } from "./seo.fixture";
import { allElementsStory } from "./all-element-story.fixture";
import { liveBlogStory } from "./live-blog-story.fixture";

export {
  allElementsStory,
  config,
  publisherConfig,
  ampConfig,
  textStory,
  storyWithManyJsEmbeds,
  relatedStories,
  configOpts,
  seo,
  liveBlogStory
};
