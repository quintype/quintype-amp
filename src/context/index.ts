// these are higher order components which wrap given component with story, config etc

import { withStory } from "./story/with-story";
import { withConfig } from "./config/with-config";
import { withStoryAndConfig } from "./story-and-config/with-story-config";

export { withStory, withConfig, withStoryAndConfig };
