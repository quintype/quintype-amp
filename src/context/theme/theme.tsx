import React from "react";
import { ThemeProvider } from "styled-components";
import defaultTokens from "./tokens";
import _merge from "lodash.merge";

import { ThemeProps } from "./theme-types";

const Theme = ({ tokens, children }: ThemeProps) => {
  const theme = tokens ? _merge({}, defaultTokens, tokens) : defaultTokens;

  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export { Theme };
