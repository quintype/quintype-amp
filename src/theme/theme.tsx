import React from "react";
import { ThemeProvider } from "styled-components";
import defaultTokens from "./tokens";

import { ThemeProps } from "./theme-types";

const Theme = ({ tokens, children }: ThemeProps) => {
  const theme = { ...defaultTokens, ...tokens };
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export { Theme };
