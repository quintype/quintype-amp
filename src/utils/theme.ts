import { AMPConfig } from "../types/config";
import get from "lodash.get";
import { invertHexColor } from "../helpers";
import defaultTokens from "../context/theme/tokens";
import merge from "lodash.merge";

export const getTokensFromAMPConfig = (config: AMPConfig) => {
  const { colors } = config;
  const { fonts } = config;
  return {
    color: {
      ...(get(colors, ["primary"], null) && { primaryColor: colors.primary }),
      ...(get(colors, ["secondary"], null) && { secondaryColor: colors.secondary }),
      ...(get(colors, ["header-background"], null) && { headerBackground: colors["header-background"] }),
      ...(get(colors, ["footer-background"], null) && { footerBackground: colors["footer-background"] }),
      ...(get(colors, ["footer-text-color"], null) && { footerTextColor: colors["footer-text-color"] }),
      ...(get(colors, ["section-text-color"], null) && { sectionTextColor: colors["section-text-color"] })
    },
    font: {
      family: {
        ...(get(fonts, ["primary", "family"], null) && { primary: fonts.primary.family }),
        ...(get(fonts, ["secondary", "family"], null) && { secondary: fonts.secondary.family })
      }
    }
  };
};

/**
 * Inverts colors for dark theme. Currently only inverts shades of gray
 * not sure how desirable the output of this will be on non-gray colors. Check before using!
 *
 */
export const getTokensForDarkTheme = (config) => {
  const tokensFromConfig = getTokensFromAMPConfig(config.ampConfig);
  const darkThemeDefaultTokens = {
    color: {
      mono1: invertHexColor(defaultTokens.color.mono1),
      mono2: invertHexColor(defaultTokens.color.mono2),
      mono3: invertHexColor(defaultTokens.color.mono3),
      mono4: invertHexColor(defaultTokens.color.mono4),
      mono5: invertHexColor(defaultTokens.color.mono5),
      mono6: invertHexColor(defaultTokens.color.mono6),
      mono7: invertHexColor(defaultTokens.color.mono7)
    }
  };
  const tokens = merge({}, darkThemeDefaultTokens, tokensFromConfig);
  return tokens;
};
