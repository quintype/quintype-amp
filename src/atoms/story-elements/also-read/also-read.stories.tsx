import React from "react";
import { storiesOf } from "@storybook/react";
import { AlsoRead } from "./also-read";
import { textStory, config } from "../../../__fixtures__";
import { Layout } from "../../layout";

const sampleAlsoReadElement = {
  description: "",
  "page-url": "/story/7f3d5bdb-ec52-4047-ac0d-df4036ec974b/element/2373433c-557d-48fb-8b32-0f2bdd65df78",
  type: "text",
  "family-id": "9b0816a1-de95-4048-bc86-2979911bcc86",
  title: "",
  id: "2373433c-557d-48fb-8b32-0f2bdd65df78",
  metadata: {
    "linked-story-id": "a7a06a3f-8d9a-450c-8eaa-387b7c8c6323",
    "linked-story": {
      headline: "LSAC Launches New Initiatives to Support Law School Enrollment Efforts in India",
      "story-content-id": "a7a06a3f-8d9a-450c-8eaa-387b7c8c6323",
      "highlighted-text": "",
      id: "a7a06a3f-8d9a-450c-8eaa-387b7c8c6323",
      "highlighted-headline": null
    }
  },
  subtype: "also-read",
  text: "LSAC Launches New Initiatives to Support Law School Enrollment Efforts in India"
};

storiesOf("AlsoRead", module)
  .addDecorator((story) => (
    <Layout story={textStory} config={config}>
      {story()}
    </Layout>
  ))
  .add("Default", () => <AlsoRead element={sampleAlsoReadElement} story={textStory} />);
