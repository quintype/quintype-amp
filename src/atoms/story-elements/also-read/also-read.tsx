import React from "react";
import { StoryElementProps } from "../types";
import get from "lodash.get";
import styled from "styled-components";
import { Spacer } from "../../spacer";
import { withStoryAndConfig } from "../../../context";

export const StyledAlsoRead = styled.div`
  display: flex;
  align-items: center;
  font-size: ${(props) => props.theme.font.size.xs};
  span {
    font-weight: bold;
  }
  a {
    color: ${(props) => props.theme.color.primaryColor};
  }

  a:hover {
    text-decoration: none;
  }
`;

export const AlsoReadBase = ({ element, story, config }: StoryElementProps) => {
  const linkedStoryId = get(element, ["metadata", "linked-story-id"]);
  const linkedStory = get(story, ["linked-stories", linkedStoryId]);
  const alsoReadRender = get(config, ["opts", "storyElementRender", "alsoReadRender"], null);

  if (!linkedStory) {
    return null;
  }

  return alsoReadRender ? (
    alsoReadRender({ story, config, element })
  ) : (
    <StyledAlsoRead>
      <Spacer token="m" align="horizontal" />
      <span>Also read: </span>
      <Spacer token="s" align="horizontal" />
      <a href={linkedStory.url}>{linkedStory.headline}</a>
      <Spacer token="m" align="horizontal" />
    </StyledAlsoRead>
  );
};
/**
 * AlsoRead is a story element
 * The look of the AlsoRead can be changed using the render prop alsoReadRender. In case alsoReadRender is passed in the config, it is rendered. Otherwise a default AlsoRead is rendered.
 *
 * @param {Object} params object containing parameters passed to the render prop
 * @param {Object} params.story story object
 * @param {Object} params.config config object
 * @param {Object} params.element the story element. This is same as the story element found in the story API response
 *
 * ```javascript
 * ...
 * ampRoutes(app, {
 *    alsoReadRender: ({ story, config, element }) => <MyCustomAlsoRead story={story} config={config} storyElement={element} />
 * })
 * ...
 * ```
 *
 * @category StoryElements
 * @module AlsoRead
 * @component
 */
export const AlsoRead = withStoryAndConfig(AlsoReadBase);
