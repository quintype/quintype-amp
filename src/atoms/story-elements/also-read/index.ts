import { AlsoRead } from "./also-read";

export { AlsoRead };
