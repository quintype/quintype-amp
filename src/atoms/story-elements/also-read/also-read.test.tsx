import React from "react";
import { BaseAlsoRead } from "./also-read";
import { shallow, mount } from "enzyme";
import { textStory } from "../../../__fixtures__";
import { Theme } from "../../../context/theme";

const sampleAlsoReadElement = {
  description: "",
  "page-url": "/story/7f3d5bdb-ec52-4047-ac0d-df4036ec974b/element/2373433c-557d-48fb-8b32-0f2bdd65df78",
  type: "text",
  "family-id": "9b0816a1-de95-4048-bc86-2979911bcc86",
  title: "",
  id: "2373433c-557d-48fb-8b32-0f2bdd65df78",
  metadata: {
    "linked-story-id": "a7a06a3f-8d9a-450c-8eaa-387b7c8c6323",
    "linked-story": {
      headline: "LSAC Launches New Initiatives to Support Law School Enrollment Efforts in India",
      "story-content-id": "a7a06a3f-8d9a-450c-8eaa-387b7c8c6323",
      "highlighted-text": "",
      id: "a7a06a3f-8d9a-450c-8eaa-387b7c8c6323",
      "highlighted-headline": null
    }
  },
  subtype: "also-read",
  text: "LSAC Launches New Initiatives to Support Law School Enrollment Efforts in India"
};

describe("Also Read", () => {
  it("should render default", () => {
    const wrapper = shallow(<BaseAlsoRead element={sampleAlsoReadElement} story={textStory} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should render with custom styles", () => {
    const wrapper = mount(
      <Theme>
        <BaseAlsoRead element={sampleAlsoReadElement} story={textStory} inlineStyles={{ color: "red" }} />
      </Theme>
    );
    expect(
      wrapper
        .find("div")
        .at(0)
        .prop("style")
    ).toStrictEqual({ color: "red" });
  });
});
