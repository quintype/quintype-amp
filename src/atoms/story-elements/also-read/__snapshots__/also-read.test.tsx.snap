// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Also Read should render default 1`] = `
<styled.div>
  <Spacer
    align="horizontal"
    token="m"
  />
  <span />
  <Spacer
    align="horizontal"
    token="s"
  />
  <a
    href="https://www.barandbench.com/news/lsac-launches-new-initiatives-to-support-law-school-enrollment-efforts-in-india"
  >
    LSAC Launches New Initiatives to Support Law School Enrollment Efforts in India
  </a>
  <Spacer
    align="horizontal"
    token="m"
  />
</styled.div>
`;
